src/Sarif.Driver/DriverUtilities.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.CodeAnalysis.Sarif.Driver
+{
+    public static class DriverUtilities
+    {
+        /// <summary>
+        /// Verifies that the output file either does not exist or is allowed to be overwritten.
+        /// </summary>
+        /// <param name="outputFilePath">
+        /// The path to the output file.
+        /// </param>
+        /// <param name="force">
+        /// true if the --force option was specified, allowing the output file to be overwritten;
+        /// otherwise false.
+        /// </param>
+        /// <param name="fileSystem">
+        /// An object that provides access to the file system.
+        /// </param>
+        public static void VerifyOutputFileCanBeCreated(string outputFilePath, bool force, IFileSystem fileSystem)
+        {
+            if (fileSystem.FileExists(outputFilePath) && !force)
+            {
+                throw new ArgumentException(
```

Comment:
>ArgumentException [](start = 26, length = 17)

ArgumentException is a bit unexpected. I take it this is the exception we typically throw as part of validation?

Reply:
No, apparently `ValidateOptions` returns a `bool` rather than throwing. As I said in another comment, I'll try to fit `VerifyOutputFileCanBeCreated` into that pattern.

---
In reply to: [311714256](https://github.com/microsoft/sarif-sdk/pull/1625#discussion_r311714256) [](ancestors = 311714256)